### PR TITLE
feat: make fixture rendering testing work with enigma-mocker

### PR DESCRIPTION
**Pull request by LiKang6688**
Nebula serve using mocked Qlik core engine (Enigma mocker) to render the visualization charts based on the testing fixture file

Testing cases using Puppeteer capture screenshots of those rendered charts and matches against the baseline to ensure our charts look as intended and present clients with expected visuals.

Visual regression Testing flow:
- Start Nebula serve and launch Puppeteer
- Iterate testing fixture files
- Create test case per testing fixture file
- Render chart based on testing fixture file in Nebula serve using Enigma mocker
- Puppeteer capture screenshot
- Compare screenshot with the baseline image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,15 +61,11 @@ jobs:
...
       - run:
           name: Rendering tests
-          command: |
-            DocId=$(curl -s --data-binary @data/apps/Executive_Dashboard.qvf http://localhost:9076/v1/apps/import | jq -r '.attributes.id')
-            APP_ID=$DocId yarn run test:rendering
-
+          command: yarn test:regression
       - store_artifacts:
-          path: ./test/rendering/artifacts
+          path: ./test/regression/artifacts
```
Comment (veinfors) [inline .circleci/config.yml:68]: These tests are still rendering tests, right? So I would prefer we keep having them under `test/rendering`

**Review by veinfors — CHANGES_REQUESTED** [review]
These tests are still rendering tests, right? So I would prefer we keep having them under test/rendering

**Comment by veinfors** [comment]
Another thing. Now we have these test cases which are basically "skipped" due to no `qlikcore/engine` being available, and those are the ones I would like us to convert into using fixtures instead. Only then can we continue delivering stuff confidently into this repo. Here is the list:

- test/rendering/artifacts/baseline/mes_2_color_dim.png
- test/rendering/artifacts/baseline/mes_2_color_dim_dim_null.png
- test/rendering/artifacts/baseline/mes_2_color_mes.png
- test/rendering/artifacts/baseline/plugin_axes.png
- test/rendering/artifacts/baseline/plugin_line.png
- test/rendering/artifacts/baseline/plugin_point.png
- test/rendering/artifacts/baseline/scatter_render_1.png
- test/rendering/artifacts/baseline/scatter_render_2.png
- test/rendering/artifacts/baseline/scatter_render_3.png
- test/rendering/artifacts/baseline/scatter_render_4.png
- test/rendering/artifacts/baseline/scatter_render_5.png
- test/rendering/artifacts/baseline/scatter_render_6.png
- test/rendering/artifacts/baseline/scatter_render_7.png
- test/rendering/artifacts/baseline/scatter_render_8.png
- test/rendering/artifacts/baseline/scatter_render_oob1.png

**Comment by LiKang6688** [comment]
> These tests are still rendering tests, right? So I would prefer we keep having them under test/rendering

Yes, sure

**Review by bao-ho — APPROVED** [review]

**Comment by LiKang6688** [comment]
> These tests are still rendering tests, right? So I would prefer we keep having them under test/rendering

Done. now test cases are converted into fixtures and are under test/rendering

**Review by veinfors — APPROVED** [review]

**Pull request merged**